### PR TITLE
feat: add more updates to "shorebird release ios" progress

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds build progress updates for known IPA build steps.

Screenshot 2024-11-15 at 12 58 11 PM

Related to https://github.com/shorebirdtech/shorebird/issues/2598

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

File: packages/shorebird_cli/lib/src/artifact_builder.dart [hunk 1]
```diff
@@ -277,30 +279,50 @@ class ArtifactBuilder {
...
+        if (update != null) {
+          buildProgress.updateDetailMessage(update);
+        }
+      });
+
+      final stderrLines = await buildProcess.stderr
+          .transform(utf8.decoder)
+          .transform(const LineSplitter())
+          .toList();
+      final stdErr = stderrLines.join('\n');
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/artifact_builder.dart]: stderr vs. stdErr?  I don't feel strongly, just looks like you used it sometimes as stdErr and sometimes stderr.

**Review by eseidel — APPROVED** [review]

**Comment by eseidel** [comment]
Bleh, sorry.  I just saw it in my inbox.  didn't notice the WIP or draft. 🤣

File: packages/shorebird_cli/lib/src/artifact_builder.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/artifact_builder.dart]: Fixed

**Review by eseidel — APPROVED** [review]
Looks great.  Thanks!

**Pull request merged**